contracts/base/DAIPool.sol

```diff
@@ -89,20 +109,93 @@ contract DAIPool is DAIPoolInterface, Initializable {
         emit DaiDeposited(msg.sender, amount);
     }
 
-    function withdrawDai(uint256 amount) external {}
+    /**
+        @notice It allows any zDAI holder to burn their zDAI tokens and withdraw their DAIs.
+        @param amount of DAI tokens to withdraw.
+     */
+    function withdrawDai(uint256 amount) external {
+        // Burn ZDAI tokens.
+        zdai.burn(amount);
+
+        // Notify ZDAI tokens were burnt/
+        lenderInfo.zDaiBurnt(msg.sender, amount);
+
+        // Transfers DAI tokens
+        daiTransfer(msg.sender, amount);
 
-    function repayDai(uint256 amount, address borrower) external {}
+        // Emit event.
+        emit DaiWithdrawn(msg.sender, amount);
+    }
 
-    function liquidationPayment(uint256 amount, address liquidator) external {}
+    /**
+        @notice It allows a borrower repaying their loan. 
+        @dev This function can be called ONLY by the Loans contract.
+        @dev It requires a DAI.approve call before calling it.
+        @dev It throws a require error if borrower called DAI.approve function before calling it.
+        @param amount in DAI tokens.
+        @param borrower address that is repaying the loan.
+     */
+    function repayDai(uint256 amount, address borrower) external isLoan(msg.sender) {
+        // Transfers DAI tokens to DAIPool.
+        daiTransferFrom(borrower, amount);
 
-    function createLoan(uint256 amount, address borrower) external returns (bool) {}
+        // Emits event.
+        emit DaiRepaid(borrower, amount);
+    }
+
+    /**
+        @notice Once a loan is liquidated, it transfers the amount in DAI tokens to the liquidator address.
+        @param amount in DAI tokens to liquidate.
+        @param liquidator address to receive the tokens.
+     */
+    function liquidationPayment(uint256 amount, address liquidator)
+        external
+        isLoan(msg.sender)
+    {
+        // Transfers DAIs to the liquidator.
+        daiTransfer(liquidator, amount);
+
+        // Emits event
+        emit PaymentLiquidated(liquidator, amount);
+    }
+
+    /**
+        @notice Once the loan is created, it transfers the amount of DAIs to the borrower.
+
+        @param amount of DAI tokens to transfer.
+        @param borrower address which will receive the DAI tokens.
+        @dev This function only can be invoked by the LoansInterface implementation.
+     */
+    function createLoan(uint256 amount, address borrower) external isLoan(msg.sender) {
+        // Transfer DAIs to the borrower.
+        daiTransfer(borrower, amount);
+    }
+
+    /**
+        @notice It gets the current DAI balance for this contract.
+        @return current DAI balance.
+     */
+    function getDaiBalance() external view returns (uint256) {
```

Comment:
This function isn't used anywhere in the contract. Is there a reason it's being kept? Might be worth deleting it now.